### ISSUE TITLE
EF Core Scaffold Doesn't Respect Output Directory

### ISSUE THREAD

**Issue by Chris-Dickerson** [dotnet/project-system#8638]
## Visual Studio Version
Visual Studio 2022 Pro

## Summary
The summary is contained in this issue: RE: https://github.com/dotnet/efcore/issues/20752
Posing the problem here as per @bricelam suggestion

Issuing the scaffold command when the output directory is anything other than standard (bin/Debug) -- the command fails with:

The specified deps.json [D:\Projects\ProjectTest\bin\Debug\net6.0-windows\ProjectTest.deps.json] does not exist

Attaching the parameter " --runtime x64" to the command, adds it as a suffix rather than a prefix... 

The specified deps.json [D:\Projects\ProjectTest\bin\Debug\net6.0-windows\x64\ProjectTest.deps.json] does not exist

## Steps to Reproduce
"C:\Program Files (x86)\dotnet\dotnet.exe" ef dbcontext scaffold "data source=D:\Projects\TestProject\TestProject.db" Microsoft.EntityFrameworkCore.Sqlite --output-dir Database\Models\System --context SystemModel --force --verbose --no-build --use-database-names --no-onconfiguring --runtime x64

## Expected Behavior
Output EF Core C# files

## Actual Behavior
The specified deps.json [D:\Projects\ProjectTest\bin\Debug\net6.0-windows\ProjectTest.deps.json] does not exist

## User Impact
Not stopped by any means, if I copy the x64\Debug folder to the bin folder, run the scaffolding command -- it generates the EF as desired.

**Comment by adamint** [comment]
@drewnoakes are we involved in this command?

**Comment by bricelam** [comment]
*(Copying some excerpts from the original issue for more context.)*

I think the issue is that you're setting it in a condition:

```xml
<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
  <OutputPath>..\bin\x64\Debug\</OutputPath>
</PropertyGroup>
```

`Platform` isn't specified when we read the property, so we get the default value.

It might work if you specify `--runtime`:

```sh
dotnet ef dbcontext scaffold --runtime win10-x64 ...
```

---


We mimic `dotnet run` which only has `--runtime`. This sets `RuntimeIdentifier` which influences `PlatformTarget`.

`Platforms` and `Platform` appear to be VS-specific properties that [don't integrate well](https://github.com/dotnet/sdk/blob/3058e86e67b30696e5eacc00c9234e9cc0033985/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props#L49-L60) with the .NET Core SDK.

In other words, `dotnet run` has the same problem as `dotnet ef`. We should follow up with [dotnet/sdk](https://github.com/dotnet/sdk) and [dotnet/project-system](https://github.com/dotnet/project-system) to see what their guidance is here.

---

So it feels like our guidance is:

1. Don't use Visual Studio's Platforms drop-down list on SDK-style projects (ones with `<Project Sdk="...">`)
2. If you do, specify a default value in the project file for SDK tools to use

It may be worth filing an issue on [dotnet/project-system](https://github.com/dotnet/project-system) to see if there's anything they can do to improve this experience. For example, by generating MSBuild fragments that play nicely with the `--runtime` option.

**Comment by Chris-Dickerson** [comment]
FWIW I removed the property group that sets the output path and end up with the same results. Adding a runtime switch simply appends the platform to the end, rather than the beginning where... something is putting the platform.

**Comment by drewnoakes** [comment]
> @drewnoakes are we involved in this command?

This seems to happen on the command line. Given that, it's not a VS issue and unlikely to be related to or fixed in this repo.

**Comment by adamint** [comment]
It seems like you need a platform set for this command to work correctly. dotnet/project-system is not involved in this behavior, please open a feedback ticket against EF Core for this.
